File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> myMap;

        for (int x : nums){
            // if number is present
            if (myMap.count(x)) return true;
            myMap.insert(x);
        }

        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # build hash map to count numbers (number : frequency)
        myDict = Counter(nums)
        # build list of list to insert numbers based on their frequency. (frequency = index)
        myList = [[] for _ in range(len(nums) + 1)]

        ret = []

        # insert numbers into list
        for key, value in myDict.items():
            myList[value].append(key)
        
        # interate list from right to left
        for subList in reversed(myList):
            for x in subList:
                ret.append(x)
                k = k - 1
                if k == 0:
                    return ret
            
            if k == 0:
                return ret
        
        